pkg/scheduler/framework/plugins/dynamicresources/structuredparameters.go

```diff
@@ -128,12 +128,10 @@ func newClaimController(logger klog.Logger, class *resourcev1alpha2.ResourceClas
 	}
 	for driverName, perDriver := range namedresourcesRequests {
 		var filter *resourcev1alpha2.NamedResourcesFilter
-		if classParameters != nil {
-			for _, f := range classParameters.Filters {
-				if f.DriverName == driverName && f.ResourceFilterModel.NamedResources != nil {
-					filter = f.ResourceFilterModel.NamedResources
-					break
-				}
+		for _, f := range classParameters.Filters {
+			if f.DriverName == driverName && f.ResourceFilterModel.NamedResources != nil {
+				filter = f.ResourceFilterModel.NamedResources
+				break
 			}
 		}
 		controller, err := namedresourcesmodel.NewClaimController(filter, perDriver.requests)
```

Comment:
Was the existing behavior equivalent in this case?

Reply:
It looks the same to me. `classParameters` is now guaranteed to not be nil (a case that I had considered, in contrast to nil claim parameters), so the if check is not needed anymore. If it had been nil before, it's now empty, so the for loop doesn't do anything, just as before.